### PR TITLE
Try prototype MP Config 2.0 based on SmallRye

### PR DESCRIPTION
**Pull request by Joseph-Cass**
Currently runs SmallRye Config 1.8.0 agains MP Config 1.4.

#build

### CODE REVIEW

**Comment by Joseph-Cass** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.config-2.0.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.org.eclipse.microprofile.config-2.0
+singleton=true
+-features=com.ibm.websphere.appserver.javax.cdi-1.2; ibm.tolerates:=2.0
+-bundles=io.openliberty.org.eclipse.microprofile.config.2.0; location:="dev/api/stable/,lib/"; mavenCoordinates="org.eclipse.microprofile.config:microprofile-config-api:2.0"
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.org.eclipse.microprofile.config-2.0.feature:5]: I think you need to add
```
 io.openliberty.microprofile.config.internal.common, \
 io.openliberty.microprofile.config.internal.serverxml, \
```

File: dev/com.ibm.ws.microprofile.config.1.1.cdi.services/bnd.bnd [hunk 2]
```diff
@@ -18,9 +18,9 @@ Bundle-Description: Microprofile Config 1.1 CDI Services
 
 Import-Package: \
         javax.enterprise.*; version="[1.1,3)", \
-        org.eclipse.microprofile.config;version="[1.0,2)", \
-        org.eclipse.microprofile.config.inject;version="[1.0,2)", \
-        org.eclipse.microprofile.config.spi;version="[1.0,2)", \
+        org.eclipse.microprofile.config;version="[1.0,3)", \
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config.1.1.cdi.services/bnd.bnd:21]: You were going to remove these changes

File: dev/com.ibm.ws.microprofile.config_repeat_tests/src/com/ibm/ws/microprofile/config/fat/repeat/RepeatConfig20EE7.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.microprofile.config_repeat_tests/src/com/ibm/ws/microprofile/config/fat/repeat/RepeatConfig20EE7.java:2]: should be 2020

File: dev/io.openliberty.microprofile.config.2.0.internal/.gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+smallrye-reactive-messaging-provider-0.0.4-sources.jar
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.config.2.0.internal/.gitignore:1]: you can delete this file

File: dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+# For more information, visit: 
+# http://washome.austin.ibm.com/xwiki/bin/view/MessagesTeam/
+#
+
+#-----------------------------------------------------------------------------------------------------------------------------
+# Messages used by Config 1.1 Impl -  range is 0000-0099, plus 9999 for the emergency message
+# Messages used by Config 1.2 Impl -  range is 0100-0199, plus 9998 for the emergency message
+# Messages used by Config 1.3 Impl -  range is 0200-0299, plus 9997 for the emergency message
+# Messages used by Config 1.4 Impl -  range is 0300-0399, plus 9996 for the emergency message
+# Messages used by Config 2.0 Impl -  range is 0400-0499, plus 9995 for the emergency message
```
Comment (tevans78) [inline dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops:28]: This file is probably named wrong ... should call it ConfigServerXML.nlsprops??
Same will apply to the common bundle.

File: dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+#-----------------------------------------------------------------------------------------------------------------------------
+
+# The {0} property was not found in the configuration.
+no.such.element.CWMCG0300E=CWMCG0300E: The {0} property was not found in the configuration.
+no.such.element.CWMCG0300E.explanation=The requested property did not exist in any of the configured Config Sources.
+no.such.element.CWMCG0300E.useraction=Add the property to one of the existing Config Sources, or add a Config Source that contains the property.
+
+failed.to.schedule.cache.invalidation.CWMCG0301E=CWMCG0301E: The MicroProfile Config cache invalidation cannot be scheduled. The exception is {0}
+failed.to.schedule.cache.invalidation.CWMCG0301E.explanation=Changes in MicroProfile Config may not be reflected in the application.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops:37]: Update 

>Changes in MicroProfile Config may not be reflected in the application.

to

>Changes in MicroProfile Config might not be reflected in the application.

File: dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+failed.to.schedule.cache.invalidation.CWMCG0301E.useraction=Review the server message.log and FFDC logs to identify the problem.
+
+#-----------------------------------------------------------------------------------------------------------------------------
+# Emergency Config 2.0 error message
+#-----------------------------------------------------------------------------------------------------------------------------
+
+# The following Config API error has occurred: {0}.
+temporary.CWMCG9996E=CWMCG9996E: The following Config API error has occurred: {0}.
```
Comment (Charlotte-Holt) [inline dev/io.openliberty.microprofile.config.internal.serverxml/resources/io/openliberty/microprofile/config/internal/serverxml/resources/Config20.nlsprops:45]: Update

>The following Config API error has occurred: {0}.

to

>The following Config API error occurred: {0}.

**Comment by andrea-pichler** [comment]
Completed the requested L2 message review

**Comment by Joseph-Cass** [comment]
#build

**Comment by Joseph-Cass** [comment]
#build

**Comment by Joseph-Cass** [comment]
#run-libby-bot

**Comment by Joseph-Cass** [comment]
#run-libby-bot

**Comment by Joseph-Cass** [comment]
#build

**Comment by Joseph-Cass** [comment]
The failed test, `testConflictConfiguration` in `com.ibm.ws.app.manager.wab.installer.fat.ConfigurableWABTests`, looks unrelated to the changes made.

**Pull request merged**